core/src/main/java/org/apache/calcite/avatica/remote/TypedValue.java

```diff
@@ -359,6 +359,9 @@ private static Object serialToJdbc(ColumnMetaData.Rep type, ColumnMetaData.Rep c
           copy.add(serialToJdbc(componentRep, null, o, calendar));
         }
       }
+      if (componentRep == null && list.size() > 0) {
+        componentRep = ((TypedValue) list.get(0)).type;
```

Comment:
If we still can not got the type, we should throw an exception, can you please add a check here ?

Reply:
I think so.